src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -50,25 +50,29 @@ GetTotalEnsembleDurations(const ServerSideStats& stats)
   EnsembleDurations result;
   for (const auto& model_stats : stats.composing_models_stat) {
     if (model_stats.second.composing_models_stat.empty()) {
-      // Success count covers all successful requests, cumulative time, queue
-      // time, compute, and cache
-      const uint64_t cnt = model_stats.second.success_count;
-      // Infer/exec counts cover compute time done in inference backends,
-      // not related to cache hit times
-      const uint64_t infer_cnt = model_stats.second.inference_count;
       // Cache hit count covers cache hits, not related to compute times
       const uint64_t cache_hit_cnt = model_stats.second.cache_hit_count;
       // cache_miss_cnt should either equal infer_cnt or be zero if
       // cache is disabled or not supported for the model/scheduler type
       const uint64_t cache_miss_cnt = model_stats.second.cache_miss_count;
 
-      result.total_queue_time_avg_us +=
-          AverageDurationInUs(model_stats.second.queue_time_ns, cnt);
+      result.total_queue_time_avg_us += AverageDurationInUs(
+          model_stats.second.queue_time_ns, model_stats.second.queue_count);
       const uint64_t compute_time = model_stats.second.compute_input_time_ns +
                                     model_stats.second.compute_infer_time_ns +
                                     model_stats.second.compute_output_time_ns;
+      if (model_stats.second.compute_input_count !=
+              model_stats.second.compute_infer_count ||
+          model_stats.second.compute_infer_count !=
+              model_stats.second.compute_output_count) {
+        throw std::runtime_error(
+            "Server side statistics compute counts must be the same.");
+      }
+      const uint64_t compute_cnt = model_stats.second.compute_input_count +
+                                   model_stats.second.compute_infer_count +
+                                   model_stats.second.compute_output_count;
```

Comment:
I misspoke earlier, we should not sum the compute counts here. The `compute_cnt` should be any one of them, and all three should be equal.

Reply:
Yes. Just use one of the counts here..
`const uint64_t compute_cnt = model_stats.second.compute_input_count`

Reply:
Switched to new calculation 👍